core/src/main/scala/chisel3/experimental/dataview/SubTypeOf.scala

```diff
@@ -0,0 +1,54 @@
+package chisel3.experimental
+
+import chisel3._
+
+import scala.language.experimental.macros
+import scala.reflect.macros.blackbox.Context
+
+trait ChiselSubTypeOf[A, B]
+
+object ChiselSubTypeOf {
+  def genChiselSubTypeOf[A: c.WeakTypeTag, B: c.WeakTypeTag](c: Context): c.Tree = {
```

Comment:
```suggestion
object ChiselSubtypeOf {
  def genChiselSubtypeOf[A: c.WeakTypeTag, B: c.WeakTypeTag](c: Context): c.Tree = {
```
Subtype is one word so we shouldn't capitalize `Type`, this also makes it consistent with `.viewAsSupertype`. Can you also name the file the same as the trait? ChiselSubtypeOf`?